plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/databinding/ProjectIdValidator.java

```diff
@@ -0,0 +1,34 @@
+package com.google.cloud.tools.eclipse.ui.util.databinding;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.core.databinding.validation.IValidator;
+import org.eclipse.core.databinding.validation.ValidationStatus;
+import org.eclipse.core.runtime.IStatus;
+
+import com.google.cloud.tools.eclipse.ui.util.Messages;
+
+public class ProjectIdValidator implements IValidator {
+  // todo: this is different from .newproject.AppEngineProjectIdValidator
```

Comment:
The other validator roughly allows for two strings of letters, digits, hyphen and dot of 1 to 100 length, separated by a colon, and the first string is optional. This is rather different from the format this validator allows.